sympy/solvers/pde.py

```diff
@@ -897,11 +897,11 @@ def pde_separate_add(eq, fun, sep):
     """
     Helper function for searching additive separable solutions.
 
-    Consider an equation of two independent variables x, y and a dependent
-    variable w, we look for the product of two functions depending on different
+    Consider an equation of two independent variables x and t and a dependent
```

Comment:
I have checked this and indeed the factor-functions of `pde_separate_mul` can depend on more than one independent variable.  Consider the two-dimensional heat equation:

    from sympy import Eq, Function, pde_separate, Derivative as D
    from sympy.abc import x, y, t

    u, U, T = map(Function, 'uUT')

    eq = Eq(D(u(x, y, t), x, 2) + D(u(x, y, t), y, 2), D(u(x, y, t), t, 2))
    pde_separate(eq, u(x, y, t), [T(t), U(x, y)])

This (correctly) returns

    [Derivative(T(t), t, t)/T(t), Derivative(U(x, y), x, x)/U(x, y) + Derivative(U(x, y), y, y)/U(x, y)]